controllers/apim/ratelimitpolicy_controller.go

```diff
@@ -537,60 +156,106 @@ func (r *RateLimitPolicyReconciler) reconcileLimits(ctx context.Context, rlp *ap
+		httpRoute.SetAnnotations(httpRouteAnnotations)
+		err := r.UpdateResource(ctx, httpRoute)
+		logger.V(1).Info("reconcileNetworkResourceBackReference: update HTTPRoute", "httpRoute", client.ObjectKeyFromObject(httpRoute), "err", err)
+		if err != nil {
+			return err
 		}
 	}
+
+	return nil
 }
 
-func (r *RateLimitPolicyReconciler) attachNetworkToStatus(ctx context.Context, network *SignalingNetwork, rlp *apimv1alpha1.RateLimitPolicy) error {
+func (r *RateLimitPolicyReconciler) reconcileWASMEnvoyFilters(ctx context.Context, rlp *apimv1alpha1.RateLimitPolicy) error {
 	logger := logr.FromContext(ctx)
-	networkKind := network.NetworkKind
-	networkName := network.NetworkName
 
-	logger.V(1).Info("attaching network to status", "network kind", networkKind, "network name", networkName)
-	rlp.Status.AddEntry(networkKind, networkName, network.Gateways)
+	httpRoute, err := r.fetchHTTPRoute(ctx, rlp)
+	if err != nil {
+		// The object should also exist
+		return err
+	}
+
+	for _, parentRef := range httpRoute.Spec.CommonRouteSpec.ParentRefs {
+		gwKey := client.ObjectKey{Name: string(parentRef.Name), Namespace: httpRoute.Namespace}
+		if parentRef.Namespace != nil {
+			gwKey.Namespace = string(*parentRef.Namespace)
+		}
+		gateway := &gatewayapiv1alpha2.Gateway{}
+		err := r.Client().Get(ctx, gwKey, gateway)
+		logger.V(1).Info("reconcileWASMEnvoyFilters: get Gateway", "gateway", gwKey, "err", err)
+		if err != nil {
+			// gateway needs to exist
+			return err
+		}
+
+		// Reconcile one EnvoyFilter per gateway
+		// Gateway API Gateway resource labels will be copied to the deployment in the automated deployment
+		// For the manual deployment, the Gateway resource labels must match deployment/pod labels or envoyfilters selector will not match
+		// https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/#automated-deployment
+		ef, err := kuadrantistioutils.WASMEnvoyFilter(rlp, gwKey, gateway.GetLabels(), httpRoute.Spec.Hostnames)
+		if err != nil {
+			return err
+		}
 
-	return r.Client().Status().Update(ctx, rlp)
+		err = r.ReconcileResource(ctx, &istio.EnvoyFilter{}, ef, kuadrantistioutils.WASMEnvoyFilterPluginMutator)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-func (r *RateLimitPolicyReconciler) detachNetworkFromStatus(ctx context.Context, network *SignalingNetwork, rlp *apimv1alpha1.RateLimitPolicy) error {
+func (r *RateLimitPolicyReconciler) fetchHTTPRoute(ctx context.Context, rlp *apimv1alpha1.RateLimitPolicy) (*gatewayapiv1alpha2.HTTPRoute, error) {
 	logger := logr.FromContext(ctx)
-	networkKind := network.NetworkKind
-	networkName := network.NetworkName
 
-	logger.V(1).Info("detaching network from status", "network kind", networkKind, "network name", networkName)
-	rlp.Status.DeleteEntry(networkKind, networkName)
+	tmpNS := rlp.Namespace
+	if rlp.Spec.TargetRef.Namespace != nil {
+		tmpNS = string(*rlp.Spec.TargetRef.Namespace)
+	}
 
```

Comment:
This will allow reference to `HTTPRoute`s in different namespace as well.

Reply:
Yes, 

As initial implementation, there is (still WIP) a validation that makes sure to support only resources in the same namespace